src/base/header.rs

```diff
@@ -505,61 +501,49 @@ impl HeaderCounts {
 
     /// Creates a header counts reference from the octets slice of a message.
     ///
-    /// The slice `message` mut be the whole message, i.e., start with the
-    /// bytes of the [`Header`](struct.Header.html).
-    ///
-    /// # Panics
-    ///
-    /// This function panics if the octets slice is shorter than 24 octets.
+    /// The slice `message` must be only the part of the message representing
+    /// the counts.
     #[must_use]
-    pub fn for_message_slice(message: &[u8]) -> &Self {
-        assert!(message.len() >= mem::size_of::<HeaderSection>());
-
-        // SAFETY: The pointer cast is sound because
-        //  - HeaderCounts has repr(transparent) and
-        //  - the slice is large enough for a HeaderSection, which contains
-        //    both a Header (which we trim) and a HeaderCounts.
-        unsafe {
-            &*((message[mem::size_of::<Header>()..].as_ptr())
-                as *const HeaderCounts)
-        }
+    pub fn for_message_chunk(message: &[u8; 8]) -> &Self {
+        // SAFETY: The transmute is sound because
+        //  - HeaderCounts has #[repr(transparent)] and
+        //  - the array has the correct size as evidenced by this (unused)
```

Comment:
I think I would like this comment to more clearly state that this transmute is only for length checking and will compile away. Even with the comment it is a bit “What?“ right now.